src/lib.rs

```diff
@@ -78,3 +78,88 @@ pub use self::db::{Database, Error};
 #[cfg(feature = "jemalloc")]
 #[global_allocator]
 static GLOBAL: Jemalloc = Jemalloc;
+
+use std::path::PathBuf;
+
+/// Python Extension
+use pyo3::{prelude::*, wrap_pyfunction};
+use storage::SecondaryStorageOptions;
+/// Open a database for user, user can specify the path of database file
+#[pyfunction]
+fn open(path: String) -> PyResult<PythonDatabase> {
+    let runtime = tokio::runtime::Builder::new_current_thread()
+        .enable_all()
+        .build()?;
+    let mut options = SecondaryStorageOptions::default_for_cli();
+    options.path = PathBuf::new().join(path);
+
+    let database = runtime.block_on(async move { Database::new_on_disk(options).await });
+    Ok(PythonDatabase { runtime, database })
+}
+
+#[pymodule]
+fn risinglight(_py: Python<'_>, m: &PyModule) -> PyResult<()> {
+    m.add_function(wrap_pyfunction!(open, m)?)?;
+    Ok(())
+}
+
+use tokio::runtime::Runtime;
+#[pyclass]
+pub struct PythonDatabase {
+    runtime: Runtime,
+    database: Database,
+}
+use pyo3::exceptions::PyException;
+
+use crate::array::Chunk;
+#[pymethods]
+impl PythonDatabase {
+    fn query(&self, py: Python<'_>, sql: String) -> PyResult<Vec<Vec<PyObject>>> {
+        let result = self
+            .runtime
+            .block_on(async { self.database.run(&sql).await });
+        match result {
+            Ok(chunks) => {
+                let mut rows = vec![];
+                for chunk in chunks {
+                    let mut table = datachunk_to_python_list(py, &chunk);
+                    rows.append(&mut table);
+                }
+                Ok(rows)
+            }
+            Err(e) => Err(PyException::new_err(e.to_string())),
+        }
+    }
+}
+use pyo3::conversion::ToPyObject;
+
+use crate::types::DataValue;
+/// Convert datachunk into Python List
+pub fn datachunk_to_python_list(py: Python, chunk: &Chunk) -> Vec<Vec<PyObject>> {
+    let mut output = vec![];
+    for data_chunk in chunk.data_chunks() {
+        for row in 0..data_chunk.cardinality() {
+            let mut row_vec = vec![];
+
+            for array in data_chunk.arrays() {
+                let s = match array.get(row) {
+                    DataValue::Null => "null".to_string().to_object(py),
+                    DataValue::Bool(v) => v.to_object(py),
+                    DataValue::Int32(v) => v.to_object(py),
+                    DataValue::Int64(v) => (v).to_object(py),
+                    DataValue::Float64(v) => v.to_object(py),
+                    DataValue::String(s) if s.is_empty() => "(empty)".to_string().to_object(py),
+                    DataValue::String(s) => s.to_object(py),
+                    DataValue::Blob(s) if s.is_empty() => "(empty)".to_string().to_object(py),
+                    DataValue::Blob(s) => s.to_string().to_object(py),
```

Comment:
I guess we don't need to deal with empty strings here.
```suggestion
                    DataValue::String(s) => s.to_object(py),
                    DataValue::Blob(s) => s.to_string().to_object(py),
```